src/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java

```diff
@@ -288,33 +288,35 @@ public RepoProvider getRepoDetails(String repoModuleName, RepoTag tag)
         checkRepoRegistered(repoModuleName);
         List<Long> ret;
         if (tag != null && tag.isGitCommitHash()) {
-            ret = MongoUtils.getProjection(jdb.getCollection(TABLE_REPO_HISTORY),
-                    String.format("{%s:#,%s:#}", FIELD_RI_MODULE_NAME, 
-                            FIELD_RH_REPO_DATA + ".gitCommitHash"), 
-                    FIELD_RH_VERSION, Long.class, repoModuleName, tag.toString());
+            ret = MongoUtils.getProjection(db.getCollection(TABLE_REPO_HISTORY),
+                    new BasicDBObject(FIELD_RI_MODULE_NAME, repoModuleName)
+                            .append(FIELD_RH_REPO_DATA + ".gitCommitHash", tag.toString()),
+                    FIELD_RH_VERSION, Long.class);
         } else {
-            String whereCondition;
+            final BasicDBObject whereCondition = new BasicDBObject(
+                    FIELD_RH_MODULE_NAME, repoModuleName);
             if (tag == null || tag.equals(RepoTag.dev)) {
-                whereCondition = String.format("{%s:#}", FIELD_RH_MODULE_NAME);
+                // do nothing
```

Comment:
Didn't know you could have empty blocks like this in Java

Reply:
Yep, I wanted to keep the logic changes minimal so I left it as is. I might factor it out in a later PR.

On the other hand, I might not.